apps/academy/src/pages/tracks/nft-solidity/5.mdx

```diff
@@ -320,62 +339,360 @@ Now that we have our application setup, we can edit the Index Page.
+  const { mint, isMinting, txData } = useMint({
+    contractAddress: CONTRACT_ADDRESS,
+    abi: TierABI.abi,
+    mintingPrice: mintingPrice,
+  });
+
+  // start mint via setting mintingPrice
+  useEffect(() => {
+    if (mintingPrice !== "0" && mint) {
+      mint()
+        .then(() => {
+          setModalShow(true);
+        })
+        .catch((e) => {
+          setModalShow(false);
+          const error = e as Error;
+          console.log("Error minting NFT", error.message);
+        })
+        .finally(() => {
+          setMintingPrice("0");
+        });
+    }
+  }, [mintingPrice, mint]);
+
+  // when mint completes, return the minted nft we want to show
+  const { mintedTokenId, latestNFTMinted } = useAwaitMintResult({
+    abi: TierABI.abi,
+    contractAddress: CONTRACT_ADDRESS,
+    userWalletAddress: address,
+  });
+
+  return (
+    <div>
+      <Head>
+        <title>NFT Minter</title>
+        <meta name="description" content="D_D Academy NFT Minter frontend integration project" />
+      </Head>
+
+      <header style={header}>
+        <h1>TierNFTs</h1>
+        <ConnectButton />
+      </header>
+
+      {isUserConnected ? (
+        <main>
+          <div style={NFTFlex}>
+            <NftCard
+              name="Tier 0"
+              imageSrc="/nfts/0_basic.svg"
+              alt="basic tier nft"
+              onClick={() => setMintingPrice("0.01")}
+              disabled={isMinting}
+            />
+
+            <NftCard
+              name="Tier 1"
+              imageSrc="/nfts/1_medium.svg"
+              alt="medium tier nft"
+              onClick={() => setMintingPrice("0.02")}
+              disabled={isMinting}
+            />
+
+            <NftCard
+              name="Tier 2"
+              imageSrc="/nfts/2_premium.svg"
+              alt="premium tier nft"
+              onClick={() => setMintingPrice("0.05")}
+              disabled={isMinting}
+            />
+          </div>
 
-// ======= COPY THE CODE BELOW =======
+          {modalShow &&
+            (txData ? (
+              <SuccessfulMint
+                contractAddress={CONTRACT_ADDRESS}
+                nftMinted={latestNFTMinted}
+                mintedTokenId={mintedTokenId}
+                onClose={() => {
+                  setModalShow(false);
+                }}
+                txData={txData}
+              />
+            ) : (
+              <Minting />
+            ))}
+        </main>
+      ) : (
+        <main>
+          <div>Please connect your wallet.</div>
+        </main>
+      )}
+    </div>
+  );
+}
 
-const header = {
+const header: CSSProperties = {
   display: "flex",
+  flexDirection: "column",
   alignItems: "center",
   justifyContent: "space-between",
   paddingTop: "20px",
   paddingLeft: "50px",
   paddingRight: "50px",
 };
 
-const NFTFlex = {
+const NFTFlex: CSSProperties = {
   display: "flex",
   alignItems: "center",
   justifyContent: "space-evenly",
   gap: "50px",
 };
+```
```

Comment:
I tried `npm run build` and `npm run dev` but wasn't able to reproduce error. Are you still seeing?

Reply:
I'll resolve, but please re-open if you're still running into.

Reply:
In the https://github.com/Developer-DAO/D_D-Academy-lesson-5-frontend-rainbowkit-starter/blob/main/components/tiernft.tsx repo, the path is `../artifacts/contracts/TierNFT.sol/TierNFT.json";` and this is this is what you have in the `tiernft.tsx` that you present in the lesson, and this will reproduce the error I submitted in this comment section.

Although for us/learner, we need `"../../artifacts/contracts/TierNFT.sol/TierNFT.json"` 

I have my editor open now, and it produces the same results. 

Please scroll up to https://github.com/Developer-DAO/academy-turbo/pull/131/files#r1612380030, you can resolve it in those lines 😁 

To be honest, there are a  couple of file paths (e.g. to the components) in `tiernft.tsx` and I don't understand why there are variances in them...

Reply:
Hey @elPiablo I replied back to your comment about `../../artifacts` vs `../artifacts` -- I think you might have the artifacts folder in a different location?

Importing of files in same directory require the `./` prefix, while the `../` are just to jump back to parent folder to then dive back down from there.